### ISSUE TITLE
1.1.5.9 daily superblock time is drifting

### ISSUE THREAD

**Issue by sunk818** [biblepay/biblepay#50]
In this chart, dots closest to 0 is best. If each PoDC daily superblock time is reset at each daily superblock, this is the difference (in minutes) to the next daily superblock daily. The daily superblock has been staying within 1 hour in either direction. Lately though, it is drifting up (daily superblock is more than 24 hours) consistently. 

![daily superblock difference resets each superblock day](https://user-images.githubusercontent.com/6989799/47448361-ac32f900-d774-11e8-950d-9d6a708e97ac.png)

In this chart, a level line is best. This chart doesn't "reset" on each daily superblock so you can see the cumulative drift. Around block 60k and 70k-77k the leveling off means the PoDC daily superblock was ideal (closest to 24 hours). 

![block time drift ideal vs actual daily superblock time](https://user-images.githubusercontent.com/6989799/47448379-b654f780-d774-11e8-83bb-12c4585171a7.png)

Data I used to create [the charts is in a Google Sheets.](https://docs.google.com/spreadsheets/d/1XRkjCvENvf0HqXvVarnjslRIaYLormV1vrLFURsu2Rk/edit?usp=sharing)

znffal made a distribution chart a few days ago as well:

![znffal](https://user-images.githubusercontent.com/6989799/47448448-e4d2d280-d774-11e8-8b7a-8301883e0515.png)

**Comment by biblepay** [comment]
With 7 minute blocks, its hard for us to have a completely accurate clock (as our blocks are solved by luck in POBH, a POW variant for CPUs), but of course we should strive for something very accurate when taking into account the law of averages over a long period.

Over a 60 day period we have : 12300 blocks.
60 days ago we solved block :  67185
Today we solved block 79485 (exactly 12,300 later from the reference beginning block).
?date-difference("h", "08-27-2018 17:45:20","10-27-2018 00:53:06") beng a difference in hours of : 
1447 hours
(this is 60.292 days) meaning that we actually ran slow by .292 days over a 60 day period -
which is a -.005% drift.

So it appears we need to speed up our master clock by .005% in the next mandatory to make up for this variance.

**Comment by sunk818** [comment]
What is the ideal daily super block distance from each other 1440 minutes, or 1435 minutes (205 x 7 minutes)?

**Comment by biblepay** [comment]
First, our superblocks do not govern our target blocks per day, our target blocks per day govern the superblocks time interval.

If you study the difficulty retarget math, you will find that you have to overshoot a change by about 10% to get it to work 'relatively close' to the target- so .005% of 60.297 is very close to what we want gross, but with the overshoot its more like .055%.  I learned this from experience.

Not sure why you are trying to represent the same math again in hours - we don't need that info.

**Comment by sunk818** [comment]
Yeah, I saw the difficulty in 7 minutes as 205 blocks is 5 minutes short of a day. If you need an integer, a 6 minute block fits nicely to a day (240 blocks) but I'm sure shorter block times has significant impact on all aspects of payment.

**Comment by biblepay** [comment]
We have seven minute blocks - that was our launch parameter.  If they end up being 7.025 they are still within the covenants of the launch because we also say 205 blocks per day.
